c7n/resources/sagemaker.py

```diff
@@ -107,6 +107,53 @@ def _augment(j):
             return list(filter(None, w.map(_augment, jobs)))
 
 
+@resources.register('sagemaker-transform-job')
+class SagemakerTransformJob(QueryResourceManager):
+
+    class resource_type(object):
+        service = 'sagemaker'
+        enum_spec = ('list_transform_jobs', 'TransformJobSummaries', None)
+        detail_spec = (
+            'describe_transform_job', 'TransformJobName', 'TransformJobName', None)
+        id = 'TransformJobArn'
```

Comment:
id should probably be TransformJobName here since its the identity parameter that its passed in for all api operations on the job.

Reply:
id for universal augment in tags.py requires arn

Reply:
it also defers to a model method called get_arns if id isn't an arn.